tools/cldr-code/src/main/java/org/unicode/cldr/util/personname/PersonNameFormatter.java

```diff
@@ -327,28 +327,63 @@ public static NamePattern from(String patternString) {
 
         private static List<NamePatternElement> parse(String patternString) {
             List<NamePatternElement> result = new ArrayList<>();
-            int position = 0; // position is at start, or after }
-            // TODO Rich handle \{, \\\{...
-            while (true) {
-                int leftCurly = patternString.indexOf('{', position);
-                if (leftCurly < 0) {
-                    if (position < patternString.length()) {
-                        result.add(new NamePatternElement(patternString.substring(position, patternString.length())));
+
+            String rawValue = "";
+            Boolean curlyStarted = false;
+            final int patternLength = patternString.length();
+            int i = 0;
+            while (i < patternLength) {
+                final Character currentCharacter = patternString.charAt(i);
+
+                switch (currentCharacter) {
+                case '\\':
+                    if (i + 1 < patternLength) {
+                        rawValue += patternString.charAt(i + 1);
+                        i += 2;
+                        continue;
```

Comment:
I wonder if we want to restrict this to just escape sequences that we actually define.  I guess right now that'd just be `\{`, `\\`, and (I guess) `\}`.  Then again, maybe there's no good reason to be so restrictive and this approach future-proofs us if we add more syntax characters,,,

Reply:
I actually do want to restrict escape sequences here. This will have an effect on toString, which is used for testing positive cases.